tock-process-console/src/ui_management/pages/setup_page/setup_page.rs

```diff
@@ -59,12 +101,43 @@ impl Component for SetupPage {
     where
         Self: Sized,
     {
+        let available_ports = match tokio_serial::available_ports() {
+            Ok(ports) => ports,
+            Err(error) => panic!("ports not found! : {}", error),
+        };
+
         let input_box = InputBox::new(state, screen_idx, action_sender.clone());
 
+        let mut board_scroll_number = 0;
+        for n in 0..available_ports.len() {
+            if available_ports[n].port_name.starts_with("/dev/ttyACM") {
+                board_scroll_number += 1;
+            }
+        }
+        let mut scrollbar_state_serial = ListState::default();
+        scrollbar_state_serial.select_first();
+
+        let mut scrollbar_state_boards = ListState::default();
+        scrollbar_state_boards.select_first();
+
+        let mut showed_serials = false;
+        let mut showed_boards = true;
```

Comment:
Make an enum instead of having two boolean values. Ideally one should always be true and the other false, but this is prone to mistakes. Instead, replace these two with an enum
```rust
enum ShowState {
    JustBoards,
    AllSerial
}